src/distributions/weighted_index.rs

```diff
@@ -130,6 +131,54 @@ impl<X: SampleUniform + PartialOrd> WeightedIndex<X> {
         })
     }
 
+    /// Reuses the weighted index by assigning a new set of weights without changing the number of
+    /// weights.
+    ///
+    /// Returns an error if:
+    ///
+    /// + the number of items in the iterator does not match the number of items used to create the
+    /// distribution;
+    /// + the iterator yields invalid values (such as `f64::NAN`);
+    /// + the weights yielded by the iterator sum up to zero.
+    ///
+    /// NOTE: If this method fails the distribution should no longer be used for sampling, because
+    /// results of sampling from it are undefined.
+    pub fn assign_new_weights<I>(&mut self, weights: I) -> Result<(), WeightedError >
```

Comment:
Method name should be `replace_weights`?

Reply:
Yeah, I think I prefer that too. 👍